### ISSUE TITLE
Question: How to group tests to execute inseparably?

### ISSUE THREAD

**Issue by alex-kulakov** [nunit/nunit#3702]
I have a base class for number of tests. It has a very long initialization process (building right database structure, filling the database with data, waiting for FT indexes to be populated) but it has a detection that database is already prepared and long init operations can be skipped. Test classes which are not descendants of the base class override the database structure for their needs.

In this case It would be beneficial to group all the descendants of the base class to inseparable execution chain. Thus, only first descendant test class would have long init process and the rest would skip it and run faster thanks to reusing already existing properly prepared database.

Is there an attribute or a property or a setting to make it?

I run tests in VS 2019's Test Explorer, NUnit versions are 2.5.4.10098 (referenced as assembly with adapter installed as extension) and 3.12.0 (with NUnit3TestAdapter 3.15.0 package)

**Comment by OsirisTerje** [comment]
Have you considered using a SetUpFixture?  https://docs.nunit.org/articles/nunit/writing-tests/attributes/setupfixture.html

**Comment by alex-kulakov** [comment]
No, I will try it.

Thank you, OsirisTerje

**Comment by alex-kulakov** [comment]
I've tried and this is not what I wanted because:

1) It requires base class and all descendants to be in the same namespace. (descendants can be in different namespaces)
2) It sets up a general initialization for all tests in the namespace (there are some tests in the same namespace which are not inherited from the base class)

If I implemented desired behavior in nunit it would be similar to this (in C#):

```
IEnumerable<NUnitTestInfo> allTestsInProject = GetTestClassesOfTheAssembly();
var groupings = allTestsInProject.GroupBy(ti => ti.GroupingKey);
foreach (var grouping in groupings) {
  foreach (var testInfo in grouping) {
    RunTest(testInfo);
  }
}
```

Say ```NUnitTestInfo.GroupingKey``` would be set by some attribute, even ```[TestFixture(Group ="LongInitTests")]```. Thus, all tests in group would run one after another.

This is what I would like to achieve.
